Minor simplifications to the implementation

gasche opened the pull request.

rgrinberg commented:
Thanks.

It seems like there are some unit test failure (`$ dune runtest)`. Do you mind checking those out?

gasche commented:
Yes, I was looking at them. There is a bad interaction between the way lexing positions are collected and the change I did with the Menhir interface, and i timed out trying to find a nice solution so I removed this commit and the testsuite now passes. (I also rebased #47 not to depend on this one.)

gasche commented:
The remaining CI failures appear unrelated to the PR itself, but rather packaging errors in other software or CI configuration issues.
